scripts/oneoff/fix-agreement-extensions.py

```diff
@@ -0,0 +1,127 @@
+"""
+
+Usage:
+    scripts/oneoff/fix-agreement-extensions.py <stage> <api_token> <download_directory> [--dry-run]
+"""
+import sys
+sys.path.insert(0, '.')
+
+import os
+import re
+
+import magic
+from docopt import docopt
+
+from dmutils.apiclient import DataAPIClient
+from dmutils.s3 import S3
+from dmscripts.env import get_api_endpoint_from_stage
+
+
+def get_all_pdfs(download_directory):
+    for root, subfolder, files in os.walk(download_directory):
+        for filename in files:
+            if filename.endswith('.pdf') and not filename.startswith('2015-11-'):
+                yield os.path.join(root, filename)
+
+
+def get_filetype(path):
+    return magic.from_file(path, mime=True)
+
+
+def is_empty(path):
+    stat = os.stat(path)
+    return stat.st_size == 0
+
+
+def get_supplier_id_from_path(path):
+    match = re.search(r'/(\d+)/', path)
+    if not match:
+        raise ValueError("Could not find supplier ID in path {}".format(path))
+    return match.group(1)
+
+
+def handle_path(client, bucket, dry_run, path):
+    if is_empty(path):
+        show_contact_details(client, dry_run, path)
+    else:
+        filetype = get_filetype(path)
+        if filetype != b"application/pdf":
+            update_file_extension(client, bucket, dry_run, path, filetype)
+
+
+def show_contact_details(client, dry_run, path):
+    supplier_id = get_supplier_id_from_path(path)
+    if dry_run:
+        print("Empty file for {} - {}".format(supplier_id, os.path.basename(path)))
+    else:
+        supplier = client.get_supplier(supplier_id)['suppliers']
+        print(
+            "Empty file for {} {}".format(
+                supplier_id,
+                supplier['contactInformation'][0]['email']))
+
+
+def get_correct_file_extension(filetype):
+    extension = {
+        b"application/zip": "zip",
+        b"image/png": "png",
+        b"image/jpeg": "jpg",
+        b'application/vnd.openxmlformats-officedocument.wordprocessingml.document': 'docx',
+    }.get(filetype)
+    if not extension:
+        raise ValueError("Unknown file type: {}".format(filetype))
+    return extension
+
+
+def get_path_in_s3(path):
+    return "{}/{}".format('g-cloud-7', path.split('/g-cloud-7/')[1])
```

Comment:
No, this is based off a synced local copy of the bucket so it's stripping off the local filesystem specific stuff as well.